smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/PublishEndToEndIT.java

```diff
@@ -0,0 +1,332 @@
+            checkPage(SC_OK);
+        }
+        rClient.deactivate(root.getPath(), SC_OK);
+        isPublishQueueEmptyOfPath(root.getPath());
+        if (isPageCheckEnabled()) {
+            checkPage(SC_NOT_FOUND);
+        }
+        
+	    return true;
+	}
+    
+	/**
+     * Checks that a GET on the page on publish has the {{expectedStatus}} in the response
+     *
+     * @throws Exception if an error occurred
+     */
+    private void checkPage(boolean skipDispatcherCache, final int...  expectedStatus) throws Exception {
+        final String path = root.getPath() + ".html";
+        log.info("Checking page {} returns status {}", adminPublish.getUrl(path), expectedStatus);
+        SlingHttpResponse res = null;
+        final List<NameValuePair> queryParams = skipDispatcherCache
+                ? Collections.singletonList(
+                new BasicNameValuePair("timestamp", String.valueOf(System.currentTimeMillis())))
+                : Collections.emptyList();
+
+        res = adminPublish.doGet(path, queryParams, Collections.emptyList());
+        if (null != res && res.getStatusLine().getStatusCode() == SC_UNAUTHORIZED) {
+            throw new AssumptionViolatedException("Publish requires auth for (SAML?). Skipping...");
+        }
+        
+        try {
+            new Polling() {
+                @Override public Boolean call() throws Exception {
+                    final List<NameValuePair> queryParams = skipDispatcherCache ?
+                        Collections.singletonList(
+                            new BasicNameValuePair("timestamp", String.valueOf(System.currentTimeMillis()))) :
+                        Collections.emptyList();
+                    adminPublish.doGet(path, queryParams, Collections.emptyList(), expectedStatus);
+                    return true;
+                }
+            }.poll(TIMEOUT_PER_TRY, 1000);
+        } catch (TimeoutException te) {
+            log.warn("Checking page {} with expected status {} failed. Please check that the connectivity to {} is proper", 
+                root.getPath(), expectedStatus, adminPublish.getUrl());
+            throw te;
+        }
+    }
+
+    private void checkPage(final int... expectedStatus) throws Exception {
+        checkPage(true, expectedStatus);
+    }
+
+    /**
+     * Returns true if the page check is to be done.
+     * @return true or false
+     */
+    protected boolean isPageCheckEnabled() {
+        return true;
+    }
+
+	private Boolean checkContentDistributionAgentExists() throws ClientException {
+		JsonNode agents = adminAuthor.doGetJson(DIST_AGENTS_PATH, 3);
+		log.info("Replication agents list: {}", agents);
+		if (agents.path(PUBLISH_DIST_AGENT).isMissingNode()) {
+		    log.warn("Default distribution agent {} is missing from the distribution list", PUBLISH_DIST_AGENT);
+		    return false;
+		}
+		return true;
+	}
+    
+    private void isPublishQueueEmptyOfPath(String replicatedPath) throws Exception {
```

Comment:
rename this to `isPublishQueueEmptyOfPath`

Reply:
Sorry couldn't notice the difference

Reply:
sry! :D Corrected the name above